pkg/lockservice/waiter.go

```diff
@@ -50,30 +50,26 @@ func acquireWaiter(txn pb.WaitTxn) *waiter {
 
 func newWaiter() *waiter {
 	w := &waiter{
-		c: make(chan notifyValue, 1),
+		status:   &atomic.Int32{},
+		refCount: &atomic.Int32{},
+		c:        make(chan notifyValue, 1),
 	}
-	w.setFinalizer()
 	w.setStatus(ready)
 	return w
 }
 
-type waiterStatus int32
-
-const (
-	ready waiterStatus = iota
-	blocking
-	notified
-	completed
-)
+func (w waiter) Name() string {
+	return "lockservice.wait"
+}
 
 // waiter is used to allow locking operations to wait for the previous
 // lock to be released if a conflict is encountered.
 type waiter struct {
 	// belong to which txn
 	txn      pb.WaitTxn
 	waitFor  [][]byte
-	status   atomic.Int32
```

Comment:
Why is this changed to a pointer?

Reply:
because we need implement:

Reply:
func (w waiter) Name() {
      return "lockservice.waiter"
}

atomic.Int32 cannot used in this method, Because noCopy.

And we use Name() string interface to get object type string as map key.  If Name() string used *waiter, we must new(waiter) to call Name() to get type string.